README.md

```diff
@@ -1,243 +1,107 @@
-var modal = pathfora.Message({
-  id: 'button-ID',
-  layout: 'button',
-  msg: 'Button text',
-  onLoad: function (event, payload) {
-    // Button has been added to the website and is visible to the user
-  }
-});
-```
+- **`gulp`** - runs the `build` tasks above and watches for any changes in the `src` directory, files are served on `localhost` port `8080`.
 
-Event is fired when the widget is rendered and visible to the user. Payload provides the `widget` data and the rendered `node` element (DOMElement).
+- **`gulp docs`** - [see below](#documentation).
 
-#### `widget.onClick(event, payload)`
+- **`gulp local`** - reads some config params from an optional local file, `.env.json` and builds and watches as with the default gulp task. This can allow you to test CSS changes locally (by default `dist/pathfora.min.js` loads the most recently deployed CSS file) or override the Lytics API URL. 
 
-```js
-var modal = pathfora.Message({
-  id: 'button-ID',
-  layout: 'button',
-  msg: 'Button text',
-  onClick: function (event, payload) {
-    // Button has been clicked
-  }
-});
-```
+  Example `.env.json` file, (using local CSS):
 
-Only `button` layout widgets.
-Event is fired when the widget button is clicked. Payload provides the `widget` data and the original click `event` payload.
-
-#### `widget.onFormSubmit(event, payload)`
-
-```js
-var modal = pathfora.Form({
-  msg: 'Subscription',
-  header: 'Header',
-  layout: 'slideout',
-  position: 'left',
-  onFormSubmit: function (event, payload) {
-    // Form has been submitted
-    // Payload:
-    // {
-    //   event: ...,
-    //   widget: ...,
-    //   data: [
-    //     { name: 'input1', value: 'value1' },
-    //     { name: 'input2', value: 'value2' },
-    //     ...
-    //   ]
-    // }
+  ```json
+  {
+    "APIURL": "https://api.lytics.io",
+    "CSSURL": "http://localhost:8080/dist/pathfora.min.css"
   }
-});
+  ```
+
+### Documentation
+Soon we will be hosting Pathfora documentation externally, for now documentation can be run locally and contributed to, using the `gulp docs` task. Our docs are powered by [mkdocs](http://www.mkdocs.org/) which you much install before running the `gulp docs` task.
+
+```sh 
+$ pip install mkdocs
+$ gulp docs
 ```
 
-Only `Form` type widgets.
-Event is fired when the form in the widget is submitted. Payload provides the `widget` data,  the original form submit `event` payload and the `data` input values from the form.
+Documentation will serve on `localhost` port `8000` while running this task.
+
+The source code for all the examples provided in the documentation live in `docs/docs/examples/src`. Preview images for the examples live in `docs/docs/examples/images`.
+
+The docs task will walk through every `.js` file in the examples source directory and compile it as a working html example in `docs/docs/examples/preview` using a handlebars template. These js files also get used are the source code to populate the `<pre>` elements within the docs.
```

Comment:
`js files also get used are the source` --> `js files also get used as the source`